pkg/didcomm/protocol/decorator/decorator.go

```diff
@@ -42,3 +42,48 @@ type Transport struct {
 type ReturnRoute struct {
 	Value string `json:"~return_route,omitempty"`
 }
+
+// Attachment intended to provide the possibility to include files, links or even JSON payload to the message.
+// To find out more please visit https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0017-attachments
+type Attachment struct {
+	// ID is a JSON-LD construct that uniquely identifies attached content within the scope of a given message.
+	// Recommended on appended attachment descriptors. Possible but generally unused on embedded attachment descriptors.
+	// Never required if no references to the attachment exist; if omitted, then there is no way
+	// to refer to the attachment later in the thread, in error messages, and so forth.
+	// Because @id is used to compose URIs, it is recommended that this name be brief and avoid spaces
+	// and other characters that require URI escaping.
+	ID string `json:"@id"`
```

Comment:
It would be a good idea to add [omitempty](https://golang.org/pkg/encoding/json/#Marshal) option to the json tag.